### PR TITLE
Add onnxruntime gpu for cmake

### PR DESCRIPTION
**Pull request by yuekaizhang**
Resolved #138

### CODE REVIEW

File: sherpa-onnx/csrc/session.cc [hunk 1]
```diff
@@ -28,15 +28,19 @@ static Ort::SessionOptions GetSessionOptionsImpl(int32_t num_threads,
...
-          // The CUDA provider is available, proceed with setting the options
-          OrtCUDAProviderOptions options;
-          options.device_id = 0;
-          // set more options on need
-          sess_opts.AppendExecutionProvider_CUDA(options);
+      std::vector<std::string> available_providers =
+          Ort::GetAvailableProviders();
+      if (std::find(available_providers.begin(), available_providers.end(),
+                    "CUDAExecutionProvider") != available_providers.end()) {
+        // The CUDA provider is available, proceed with setting the options
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/session.cc:36]: what if we don't check it?

Comment (yuekaizhang) [reply sherpa-onnx/csrc/session.cc:36]: Then it would be a segmentation error. If we use cpu build but --provider=gpu.

**Comment by csukuangfj** [comment]
Thanks! Is it ready to merge?

**Comment by yuekaizhang** [comment]
> Thanks! Is it ready to merge?

Yes. It's ready.

**Pull request merged**